### DISCUSSION TITLE
How do I get the list of all collections a user has access to in my custom endpoint?

### DISCUSSION THREAD

**Discussion by sarthakgaur** [directus/directus#7402] [Archive]
So, in my custom module's Vue component I can do this `this.system.api.get("/collections/")` to get the list of all collections this user has access to. I need access to this same list on my custom endpoint. How can I do it? Any help will be appreciated! Thanks!

**Comment by sarthakgaur** [comment]
Is this the right way to do it?

```
const collectionsService = new CollectionsService({
    schema: req.schema,
    accountability: req.accountability,
});

const allCollections = await collectionsService.readByQuery();
```

**Reply by rijkvanzanten** [reply]
Yep! That's exactly the right way to do it 👍🏻
